extensions-core/s3-extensions/src/main/java/org/apache/druid/data/input/s3/S3InputSource.java

```diff
@@ -87,6 +91,6 @@ public String toString()
 
   private Iterable<S3ObjectSummary> getIterableObjectsFromPrefixes()
   {
-    return () -> S3Utils.objectSummaryIterator(s3Client, getPrefixes(), MAX_LISTING_LENGTH);
+    return () -> S3Utils.objectSummaryIterator(s3Client, getPrefixes(), segmentPusherConfig.getMaxListingLength());
```

Comment:
`MAX_LISTING_LENGTH` is defined in the parent class (`CloudObjectInputSource`) and is not used anymore. Please remove it.

Reply:
fixed